### PR TITLE
Issue 888

### PR DESCRIPTION
**Pull request by youming-lin**
https://github.com/IBM-Swift/Kitura/issues/888

### CODE REVIEW

**Comment by youming-lin** [comment]
@ianpartridge or @shmuelk Can you review this?

**Comment by irar2** [comment]
@youming-lin You didn't improve the basic `UserProfile` creation, you only pass an id and name. Was this intentional? 
The idea was to fill in as much as possible by the plugin leaving to the user only the `extendedProperties` (see `CredentialsGoogle.Utils.createUserProfile` for example).

**Comment by youming-lin** [comment]
@irar2 I have updated code with additional fields to `UserProfile`.

File: Sources/CredentialsGitHub/CredentialsGitHub.swift [hunk 1]
```diff
@@ -123,9 +126,12 @@ public class CredentialsGitHub : CredentialsPluginProtocol {
                                         try profileResponse.readAllData(into: &body)
                                         jsonBody = JSON(data: body)
 
-                                        if let id = jsonBody["id"].number?.stringValue {
-                                            let name = jsonBody["name"].stringValue
-                                            let userProfile = UserProfile(id: id, displayName: name, provider: self.name)
+                                        if let userProfile = self.createUserProfile(from: jsonBody) {
```
Comment (irar2) [inline Sources/CredentialsGitHub/CredentialsGitHub.swift]: Please convert jsonBody to a dictionary and change createUserProfile to accept a dictionary.
(We may be re-evaluating our use of SwiftyJSON in APIs).

**Comment by youming-lin** [comment]
@irar2 I have changed code to use `[String: Any]` instead of `JSON`.

**Comment by irar2** [comment]
Thanks @youming-lin !

**Pull request merged**